packages/react/src/components/Accordion/Accordion.tsx

```diff
@@ -146,33 +197,73 @@ class Accordion extends AutoControlledComponent<WithAsProp<AccordionProps>, any>
       _.invoke(predefinedProps, 'onClick', e, titleProps)
       _.invoke(this.props, 'onTitleClick', e, titleProps)
     },
+    onFocus: (e: React.SyntheticEvent, titleProps: AccordionTitleProps) => {
+      _.invoke(predefinedProps, 'onFocus', e, titleProps)
+      this.setState({ focusedIndex: titleProps.index })
+    },
   })
 
-  isIndexActive = (index): boolean => {
+  private isIndexActive = (index: number): boolean => {
     const { exclusive } = this.props
     const { activeIndex } = this.state
 
     return exclusive ? activeIndex === index : _.includes(activeIndex, index)
   }
 
+  private isIndexClosingPrevented = (index: number): boolean => {
+    const { activeIndex } = this.state
+    const { alwaysActive, exclusive } = this.props
+
+    if (alwaysActive) {
+      if (exclusive) {
+        return activeIndex === index
+      }
+      return activeIndex.length === 1 && activeIndex[0] === index
```

Comment:
What should happen if the `alwaysActive` is used with `exclusive = false`, and the index is not the first element?

Reply:
actually my logic here is stupid. I will refactor here and hopefully it will not be stupid. again.